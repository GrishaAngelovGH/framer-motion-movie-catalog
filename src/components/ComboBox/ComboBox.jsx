const ComboBox = ({ id, title, products, onAddToCart }) => {
  const totalPrice = products.reduce((a, b) => a + b.price, 0)

  return (
    <div className="bg-white rounded-md mt-10 text-center min-w-[300px]">
      <div className="text-white  p-1 bg-blue-400 rounded-t">{title}</div>
      <div className="flex flex-col md:flex-row justify-evenly p-5">
        {
          products.map((v, i) => (
            <div key={i} className="m-3">
              <img src={v.image} className="inline w-[100px] h-[100px] border border-gray-600 rounded-full p-2" />
              <div className="mt-2 text-gray-500">{v.title} ({v.price}£)</div>
            </div>
          ))
        }
      </div>
      <p className="text-xl">Total: {totalPrice}£</p>
      <button
        onClick={() => { onAddToCart(id) }}
        className="bg-blue-500 hover:bg-blue-700 text-white rounded-full m-3"
      >
        Add To Cart
      </button>
    </div>
  )
}

export default ComboBox