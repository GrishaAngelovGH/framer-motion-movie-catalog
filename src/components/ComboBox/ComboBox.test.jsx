import { render, screen, fireEvent } from "@testing-library/react"
import { expect, vi } from "vitest"

import ComboBox from "./ComboBox"

test("should render ComboBox component", () => {
  const view = render(
    <ComboBox
      title="Combo Box Title"
      products={[{ image: "image-1.jpg", title: "Product Title", price: 5 }]}
    />
  )

  expect(view).toMatchSnapshot()
})

test("should render ComboBox component with added items to shopping cart", () => {
  const onAddToCart = vi.fn()

  const view = render(
    <ComboBox
      title="Combo Box Title"
      id={123}
      products={[{ image: "image-1.jpg", title: "Product Title", price: 5 }]}
      onAddToCart={onAddToCart}
    />
  )

  const AddToCartButton = screen.getByRole("button", { name: /Add To Cart/i })

  fireEvent.click(AddToCartButton)

  expect(view).toMatchSnapshot()
  expect(onAddToCart).toBeCalledTimes(1)
  expect(onAddToCart).toBeCalledWith(123)
})