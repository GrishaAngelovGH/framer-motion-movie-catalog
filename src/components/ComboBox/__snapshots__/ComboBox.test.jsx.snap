// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render ComboBox component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded-md mt-10 text-center min-w-[300px] relative"
      >
        <div
          class="text-white  p-1 bg-blue-400 rounded-t"
        >
          Combo Box Title
        </div>
        <div
          class="flex flex-col md:flex-row justify-evenly p-5"
        >
          <div
            class="m-3"
          >
            <img
              class="inline w-[100px] h-[100px] border border-gray-600 rounded-full p-2"
              src="image-1.jpg"
            />
            <div
              class="mt-2 text-gray-500"
            >
              Product Title
               (
              5
              £)
            </div>
          </div>
        </div>
        <p
          class="text-xl"
        >
          Total: 
          5
          £
        </p>
        <button
          class="bg-blue-500 hover:bg-blue-700 text-white rounded-full m-3 visible"
          role="button"
        >
          Add To Cart
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded-md mt-10 text-center min-w-[300px] relative"
    >
      <div
        class="text-white  p-1 bg-blue-400 rounded-t"
      >
        Combo Box Title
      </div>
      <div
        class="flex flex-col md:flex-row justify-evenly p-5"
      >
        <div
          class="m-3"
        >
          <img
            class="inline w-[100px] h-[100px] border border-gray-600 rounded-full p-2"
            src="image-1.jpg"
          />
          <div
            class="mt-2 text-gray-500"
          >
            Product Title
             (
            5
            £)
          </div>
        </div>
      </div>
      <p
        class="text-xl"
      >
        Total: 
        5
        £
      </p>
      <button
        class="bg-blue-500 hover:bg-blue-700 text-white rounded-full m-3 visible"
        role="button"
      >
        Add To Cart
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render ComboBox component with added items to shopping cart 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded-md mt-10 text-center min-w-[300px] relative"
      >
        <div
          class="text-white  p-1 bg-blue-400 rounded-t"
        >
          Combo Box Title
        </div>
        <div
          class="flex flex-col md:flex-row justify-evenly p-5"
        >
          <div
            class="m-3"
          >
            <img
              class="inline w-[100px] h-[100px] border border-gray-600 rounded-full p-2"
              src="image-1.jpg"
            />
            <div
              class="mt-2 text-gray-500"
            >
              Product Title
               (
              5
              £)
            </div>
          </div>
        </div>
        <p
          class="text-xl"
        >
          Total: 
          5
          £
        </p>
        <button
          class="bg-blue-500 hover:bg-blue-700 text-white rounded-full m-3 invisible"
          role="button"
        >
          Add To Cart
        </button>
        <div
          class="bg-green-600 text-white rounded-full p-3 absolute bottom-3 left-[20%] md:left-[32%] xl:left-[38%] text-center"
        >
          Succesfully added
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded-md mt-10 text-center min-w-[300px] relative"
    >
      <div
        class="text-white  p-1 bg-blue-400 rounded-t"
      >
        Combo Box Title
      </div>
      <div
        class="flex flex-col md:flex-row justify-evenly p-5"
      >
        <div
          class="m-3"
        >
          <img
            class="inline w-[100px] h-[100px] border border-gray-600 rounded-full p-2"
            src="image-1.jpg"
          />
          <div
            class="mt-2 text-gray-500"
          >
            Product Title
             (
            5
            £)
          </div>
        </div>
      </div>
      <p
        class="text-xl"
      >
        Total: 
        5
        £
      </p>
      <button
        class="bg-blue-500 hover:bg-blue-700 text-white rounded-full m-3 invisible"
        role="button"
      >
        Add To Cart
      </button>
      <div
        class="bg-green-600 text-white rounded-full p-3 absolute bottom-3 left-[20%] md:left-[32%] xl:left-[38%] text-center"
      >
        Succesfully added
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
